### ISSUE TITLE
Rev4200 & ZeroMail (http://127.0.0.1:43110/Mail.ZeroNetwork.bit/)

### ISSUE THREAD

**Issue by katian** [ZeroNerds/zeronet-snap#8]
Internal error: AttributeError: /snap/zeronet/2019/usr/lib/x86_64-linux-gnu/libssl.so.1.1: undefined symbol: ECDH_OpenSSL
UiWebsocket.py line 79 > UiWebsocket.py line 232 > CryptMessage/CryptMessagePlugin.py line 86 > __init__.py line 17 > ecc.py line 15 > cipher.py line 7 > __init__.py line 43 > openssl.py line 513 > openssl.py line 168 > __init__.py line 361 > __init__.py line 366

**Comment by mkg20001** [comment]
When does this happen exactly?
When visiting the site, I don't get that problem

**Comment by katian** [comment]
it works now thanks mkg20001

**Comment by katian** [comment]
Hi, again, when i open http://127.0.0.1:43110/Mail.ZeroNetwork.bit/ :

Internal error: AttributeError: /snap/zeronet/2115/usr/lib/x86_64-linux-gnu/libssl.so.1.1: undefined symbol: ECDH_OpenSSL
UiWebsocket.py line 79 > UiWebsocket.py line 232 > CryptMessage/CryptMessagePlugin.py line 86 > __init__.py line 17 > ecc.py line 15 > cipher.py line 7 > __init__.py line 43 > openssl.py line 513 > openssl.py line 168 > __init__.py line 361 > __init__.py line 366

snap installed:   0.7.1-r4210            (2115)

https://github.com/HelloZeroNet/ZeroMail/issues/43#issuecomment-531479667

**Comment by mkg20001** [comment]
Seems like it has re-appeared

@HelloZeroNet do you have a clue what's going on? Could some libraries be missing from the snap?

**Comment by HelloZeroNet** [comment]
TLDR: Fixed in Rev4214 
https://github.com/HelloZeroNet/ZeroNet/commit/10817aefae60b84d25228878642d840a91188340
https://github.com/HelloZeroNet/ZeroNet/commit/4293a44c93d4fef853a3df47064843f7278f3030
https://github.com/HelloZeroNet/ZeroNet/commit/b474677db1ff06a237206a7aa43cc52eb6ffce98

It took me a while to figure out how to debug snap package (maybe there is a better way to do it), here is how I done (maybe it could be helpful to others):
 - Installed snapd on debian VPS
 - snap install zeronet
 - snap download zeronet
 - unsquashfs zeronet_2115.snap
 - mount --bind -o nodev,ro /root/squashfs-root/ /snap/zeronet/current
 - Modified /root/squashfs-root/src/main.py start() to make it easier to debug:
```py
def start():
    from lib import pyelliptic
    print("Loaded.")
    return False
```
- Error: `OSError: libssl.so.1.0.2: cannot open shared object file: No such file or directory`
- Removed that from util/OpenSslFindPatch.py
- `AttributeError: /snap/zeronet/2115/usr/lib/x86_64-linux-gnu/libssl.so.1.1: undefined symbol: ECDH_OpenSSL`
- After a while some testing I figured out it was loading the pyelliptic lib from site-packages that is not compatible with Openssl 1.1

**Comment by mkg20001** [comment]
You can use `snap try <directory>` from the "prime" directory after running `snapcraft` to build the snap using the files from this repo

**Comment by HelloZeroNet** [comment]
I wanted to avoid rebuilding as it could mean different version of the libraries.

**Comment by mkg20001** [comment]
`snapcraft` spawns a vm via multipass if you are not using the specific distro the snap needs to be compiled on

But after unsquashing it `snap try <folder>` should've worked as well

**Comment by mkg20001** [comment]
Build should be released automatically in the next 8 hours.

@katian Can you close the issue once it works?

**Comment by katian** [comment]
snap installed:   0.7.1-r4214            (2133) 21MB 

thats great, it works !

thanks a lot !
